stl/inc/yvals_core.h

```diff
@@ -414,14 +414,119 @@
+                   "It is not useful to call this method and discard the returned value. "                      \
+                   "There's no way to clear an array as its size is fixed")
+
+#define _NODISCARD_BARRIER_TOKEN \
+    _NODISCARD_MSG("The token from 'arrive()' should not be discarded; it should be passed to 'wait()'")
+
+#define _NODISCARD_TRY_WAIT                                                                                            \
+    _NODISCARD_MSG("This method returns the state of the synchronization object and does not do anything else; it is " \
+                   "not useful to call this method and discard the return value")
+
+#define _NODISCARD_TRY_CHANGE_STATE                                                                                   \
+    _NODISCARD_MSG("This method returns whether the operation succeeds in modifying the state of the object or not. " \
+                   "It is dangerous to ignore the return value")
+
+#define _NODISCARD_TRY_CHANGE_STATE_FN                                                                          \
+    _NODISCARD_MSG("This function returns whether the operation succeeds in modifying the state of the passed " \
+                   "objects or not. It is dangerous to ignore the return value")
+
+#ifdef _NODISCARD_LOCK_SUPPRESS
+
+#define _NODISCARD_LOCK
+#define _NODISCARD_CTOR_LOCK
+
+#else // ^^^ defined(_NODISCARD_LOCK_SUPPRESS) ^^^ / vvv !defined(_NODISCARD_LOCK_SUPPRESS) vvv
+
+#define _NODISCARD_LOCK                                                                                              \
+    _NODISCARD_MSG(                                                                                                  \
+        "A lock should be saved in a variable to protect the scope. (If the intetion is to protect the rest of the " \
+        "current statement, using comma operator, please use cast to void to suppress this warning. "                \
+        "Alternatively, define _NODISCARD_LOCK_SUPPRESS.)")
+
+#define _NODISCARD_CTOR_LOCK                                                                                         \
+    _NODISCARD_CTOR_MSG(                                                                                             \
+        "A lock should be saved in a variable to protect the scope. (If the intetion is to protect the rest of the " \
+        "current statement, using comma operator, please use cast to void to suppress this warning)"                 \
+        "Alternatively, define _NODISCARD_LOCK_SUPPRESS.)")
+
+#endif // ^^^ !defined(_NODISCARD_LOCK_SUPPRESS) ^^^
+
+#define _NODISCARD_CTOR_THREAD \
+    _NODISCARD_CTOR_MSG("Thread is not joined or detached, terminate will be called at the end of the statement")
+
+#define _NODISCARD_CTOR_JTHREAD \
+    _NODISCARD_CTOR_MSG("Thread is implicitly joined at the end of the statement, is this an intention?")

```

Comment:
While I agree `std::thread` constructor should be `[[nodiscard]]`, I'm questioning it for `std::jtrhead` constructor.

Reply:
It can be suppressed with `jthread(...).join()`